File: doc-examples/example-bson-java/src/main/java/example/Book.java
package example;

import io.micronaut.serde.annotation.Serdeable;
import org.bson.codecs.pojo.annotations.BsonCreator;
import org.bson.codecs.pojo.annotations.BsonProperty;

@Serdeable // <1>
public class Book {
    private final String title;
    @BsonProperty("qty") // <2>
    private final int quantity;

    @BsonCreator // <3>
    public Book(String title, int quantity) {
        this.title = title;
        this.quantity = quantity;
    }

    public String getTitle() {
        return title;
    }

    public int getQuantity() {
        return quantity;
    }
}

File: doc-examples/example-bson-kotlin/src/main/kotlin/example/Book.kt
package example

import io.micronaut.serde.annotation.Serdeable
import org.bson.codecs.pojo.annotations.BsonCreator
import org.bson.codecs.pojo.annotations.BsonProperty

@Serdeable // <1>
class Book @BsonCreator constructor( // <3>
    val title: String,
    @field:BsonProperty("qty") val quantity: Int // <2>
)
